cmd/telemeter-server/main.go

```diff
@@ -317,28 +316,38 @@ func (o *Options) Run() error {
 	internalPaths := []string{"/", "/federate", "/metrics", "/debug/pprof", "/healthz", "/healthz/ready"}
 
 	// configure the authenticator and incoming data validator
-	auth := server.New(o.PartitionKey, authorizeURL, authorizeClient, o.TokenExpireSeconds, signer, o.Labels)
-	validator := untrusted.NewValidator(o.PartitionKey, o.Labels, o.LimitBytes, 24*time.Hour)
+	var clusterAuth authorize.ClusterAuthorizer
+	clusterAuth = authorize.ClusterAuthorizerFunc(stub.Authorize)
+	if authorizeURL != nil {
+		clusterAuth = tollbooth.NewAuthorizer(authorizeClient, authorizeURL)
+	}
+
+	auth := jwt.NewAuthorizeClusterHandler(o.PartitionKey, o.TokenExpireSeconds, signer, o.Labels, clusterAuth)
```

Comment:
in a follow up we really need to slim down this tremendous `Run` method

Reply:
yes :+1: